diff --git a/op-deployer/pkg/deployer/bootstrap/flags.go b/op-deployer/pkg/deployer/bootstrap/flags.go
index e55bc204ed..bdc0121bba 100644
--- a/op-deployer/pkg/deployer/bootstrap/flags.go
+++ b/op-deployer/pkg/deployer/bootstrap/flags.go
@@ -153,6 +153,7 @@ var ImplementationsFlags = []cli.Flag{
 	ProofMaturityDelaySecondsFlag,
 	DisputeGameFinalityDelaySecondsFlag,
 	SuperchainConfigProxyFlag,
+	deployer.CacheDirFlag,
 	ProtocolVersionsProxyFlag,
 	UpgradeControllerFlag,
 	UseInteropFlag,
@@ -163,6 +164,7 @@ var ProxyFlags = []cli.Flag{
 	deployer.PrivateKeyFlag,
 	OutfileFlag,
 	deployer.ArtifactsLocatorFlag,
+	deployer.CacheDirFlag,
 	ProxyOwnerFlag,
 }
 
@@ -175,6 +177,7 @@ var SuperchainFlags = []cli.Flag{
 	ProtocolVersionsOwnerFlag,
 	GuardianFlag,
 	PausedFlag,
+	deployer.CacheDirFlag,
 	RequiredProtocolVersionFlag,
 	RecommendedProtocolVersionFlag,
 }
diff --git a/op-deployer/pkg/deployer/broadcaster/gas_estimator.go b/op-deployer/pkg/deployer/broadcaster/gas_estimator.go
index b04390fc8a..f81e8eb888 100644
--- a/op-deployer/pkg/deployer/broadcaster/gas_estimator.go
+++ b/op-deployer/pkg/deployer/broadcaster/gas_estimator.go
@@ -12,14 +12,14 @@ var (
 	// baseFeePadFactor = 50% as a divisor
 	baseFeePadFactor = big.NewInt(2)
 	// tipMulFactor = 5 as a multiplier
-	tipMulFactor = big.NewInt(5)
+	tipMulFactor = big.NewInt(1)
 	// dummyBlobFee is a dummy value for the blob fee. Since this gas estimator will never
 	// post blobs, it's just set to 1.
 	dummyBlobFee = big.NewInt(1)
 	// maxTip is the maximum tip that can be suggested by this estimator.
 	maxTip = big.NewInt(50 * 1e9)
 	// minTip is the minimum tip that can be suggested by this estimator.
-	minTip = big.NewInt(1 * 1e9)
+	minTip = big.NewInt(1 * 1e7)
 )
 
 // DeployerGasPriceEstimator is a custom gas price estimator for use with op-deployer.
diff --git a/op-deployer/pkg/deployer/flags.go b/op-deployer/pkg/deployer/flags.go
index 5ec0b2c4c8..e6edf0ca95 100644
--- a/op-deployer/pkg/deployer/flags.go
+++ b/op-deployer/pkg/deployer/flags.go
@@ -159,6 +159,7 @@ var ApplyFlags = []cli.Flag{
 	WorkdirFlag,
 	PrivateKeyFlag,
 	DeploymentTargetFlag,
+	CacheDirFlag,
 }
 
 var UpgradeFlags = []cli.Flag{
diff --git a/op-deployer/pkg/deployer/standard/standard.go b/op-deployer/pkg/deployer/standard/standard.go
index a102959f60..56e9072a55 100644
--- a/op-deployer/pkg/deployer/standard/standard.go
+++ b/op-deployer/pkg/deployer/standard/standard.go
@@ -17,23 +17,27 @@ import (
 )
 
 const (
-	GasLimit                        uint64 = 60_000_000
-	BasefeeScalar                   uint32 = 1368
-	BlobBaseFeeScalar               uint32 = 801949
-	WithdrawalDelaySeconds          uint64 = 302400
+	GasLimit               uint64 = 60_000_000
+	BasefeeScalar          uint32 = 1368
+	BlobBaseFeeScalar      uint32 = 801949
+	WithdrawalDelaySeconds uint64 = 60
+	// WithdrawalDelaySeconds          uint64 = 302400
 	MinProposalSizeBytes            uint64 = 126000
-	ChallengePeriodSeconds          uint64 = 86400
-	ProofMaturityDelaySeconds       uint64 = 604800
-	DisputeGameFinalityDelaySeconds uint64 = 302400
-	MIPSVersion                     uint64 = 2
-	DisputeGameType                 uint32 = 1 // PERMISSIONED game type
-	DisputeMaxGameDepth             uint64 = 73
-	DisputeSplitDepth               uint64 = 30
-	DisputeClockExtension           uint64 = 10800
-	DisputeMaxClockDuration         uint64 = 302400
-	Eip1559DenominatorCanyon        uint64 = 250
-	Eip1559Denominator              uint64 = 50
-	Eip1559Elasticity               uint64 = 6
+	ChallengePeriodSeconds          uint64 = 60
+	ProofMaturityDelaySeconds       uint64 = 120
+	DisputeGameFinalityDelaySeconds uint64 = 60
+	// ChallengePeriodSeconds          uint64 = 86400
+	// ProofMaturityDelaySeconds       uint64 = 604800
+	// DisputeGameFinalityDelaySeconds uint64 = 302400
+	MIPSVersion              uint64 = 2
+	DisputeGameType          uint32 = 1 // PERMISSIONED game type
+	DisputeMaxGameDepth      uint64 = 73
+	DisputeSplitDepth        uint64 = 30
+	DisputeClockExtension    uint64 = 30
+	DisputeMaxClockDuration  uint64 = 60
+	Eip1559DenominatorCanyon uint64 = 250
+	Eip1559Denominator       uint64 = 50
+	Eip1559Elasticity        uint64 = 6
 
 	ContractsV160Tag        = "op-contracts/v1.6.0"
 	ContractsV180Tag        = "op-contracts/v1.8.0-rc.4"
@@ -155,6 +159,8 @@ func SuperchainProxyAdminAddrFor(chainID uint64) (common.Address, error) {
 		return common.HexToAddress("0x543bA4AADBAb8f9025686Bd03993043599c6fB04"), nil
 	case 11155111:
 		return common.HexToAddress("0x189aBAAaa82DfC015A588A7dbaD6F13b1D3485Bc"), nil
+	case 2345:
+		return common.HexToAddress("0x6f89b7efe115e68628a6887713dcf104c15a3e51"), nil
 	default:
 		return common.Address{}, fmt.Errorf("unsupported chain ID: %d", chainID)
 	}
